src/Dev/Tasks/FileMigrationHelper.php

```diff
@@ -240,7 +263,11 @@ protected function migrateFile($base, File $file, $legacyFilename)
 
         // Save and publish
         try {
-            $file->write();
+            if (class_exists(Versioned::class)) {
+                $file->writeToStage(Versioned::LIVE);
```

Comment:
This should actually speed up the process because, the old logic would write to file to the protected store and write it back to the public store. This one leaves the file where it is.